jerry-core/ecma/base/ecma-globals.h

```diff
@@ -74,6 +74,7 @@ typedef enum
   ECMA_TYPE_STRING = 2, /**< pointer to description of a string */
   ECMA_TYPE_OBJECT = 3, /**< pointer to description of an object */
   ECMA_TYPE_ERROR = 7, /**< pointer to description of an error reference */
+  ECMA_TYPE_COLLECTION_CHUNK = ECMA_TYPE_ERROR, /**< pointer to description of a collection chunk */
   ECMA_TYPE___MAX = ECMA_TYPE_ERROR /** highest value for ecma types */
```

Comment:
ECMA_TYPE___MAX and ECMA_TYPE_COLLECTION_CHUNK have the same value. Is this intentional? IMHO it is confusing.

Reply:
Yes. The error is used only by the external API, while collection is used only internally. Currently I don't want to waste a type on collection.